cmd/oras/internal/option/remote.go

```diff
@@ -144,8 +145,12 @@ func (opts *Remote) NewRepository(reference string, common Common) (repo *remote
 	if err != nil {
 		return nil, err
 	}
-
-	repo.PlainHTTP = opts.PlainHTTP
+	switch host, _, _ := net.SplitHostPort(repo.Reference.Registry); host {
+	case "localhost":
+		repo.PlainHTTP = true
+	default:
+		repo.PlainHTTP = opts.PlainHTTP
+	}
```

Comment:
`NewRegistry()` also requires the same logic for `localhost`. Recommend to refactor.